numba/tests/test_function_type.py

```diff
@@ -917,14 +917,15 @@ def foo_template(funcs, x):
                 r = f(r)
             return r
 
-        # Problem:
-        a = jit(nopython=True)(lambda x: x + 1)
-        b = jit(nopython=True)(lambda x: x + 2)
-        foo = jit(nopython=True)(foo_template)
-        r = foo((a, b), 0)
-        self.assertEqual(r, 3)
-        # the Tuple type of foo first argument is UndefinedFunctionType:
-        self.assertEqual(foo.signatures[0][0].dtype.is_precise(), False)
+        if 0:
+            # Problem:
+            a = jit(nopython=True)(lambda x: x + 1)
+            b = jit(nopython=True)(lambda x: x + 2)
+            foo = jit(nopython=True)(foo_template)
+            r = foo((a, b), 0)
+            self.assertEqual(r, 3)
+            # the Tuple type of foo first argument is UndefinedFunctionType:
+            self.assertEqual(foo.signatures[0][0].dtype.is_precise(), False)
```

Comment:
Perhaps comment this block along with some explanation of why it's not going to work opposed to `if 0:` ?

Reply:
I have updated the statement of the problem that is about interpreting certain dispatcher types as precise (compilation is disabled and there exists exactly one compiled overload).
The code in `if 0:` will not work because of the disabled support of using dispatcher instances as tuple items. So, removed `if 0:` block.